### PR TITLE
Added require biallelic to docs

### LINKED ISSUE #1923: document biallelic-only methods

### PR DESCRIPTION
**Pull request by jigold**
Resolves #1923

### CODE REVIEW

File: python/hail/dataset.py [hunk 1]
```diff
@@ -1106,6 +1106,8 @@ def concordance(self, right):
 
         Requires the row key (variant) schema is :py:class:`~hail.expr.TVariant` and the genotype schema is :py:class:`~hail.expr.TGenotype`.
 
+        Requires :py:class:`~hail.VariantDataset.was_split` equals True. Use :py:meth:`~hail.VariantDataset.split_multi` to convert to a biallelic dataset.
```
Comment (tpoterba) [inline python/hail/dataset.py]: I think this message could be better. Something like:
```markdown
Requires the dataset to contain no multiallelics (``vds.was_split == True``). 
Use :py:meth:`~hail.VariantDataset.split_multi` to split multiallelic sites 
or :py:meth:`~hail.VariantDataset.filter_multi` to remove them.

File: python/hail/dataset.py [hunk 2]
```diff
@@ -1221,6 +1223,10 @@ def de_novo(self, pedigree, pop_frequency_prior, min_gq=20, min_p=0.05,
                 max_parent_ab=0.05, min_child_ab=0.20, min_depth_ratio=0.10):
         """Call de novo variation from trio data.
 
+        Requires the row key (variant) schema is :py:class:`~hail.expr.TVariant` and the genotype schema is :py:class:`~hail.expr.TGenotype`.
```
Comment (tpoterba) [inline python/hail/dataset.py]: Replace the "is" with "to be"

File: src/main/scala/is/hail/variant/VariantDataset.scala [hunk 3]
```diff
@@ -436,7 +436,7 @@ class VariantDatasetFunctions(private val vds: VariantDataset) extends AnyVal {
   }
 
   def linreg(ys: Array[String], covariates: Array[String] = Array.empty[String], root: String = "va.linreg", useDosages: Boolean = false, variantBlockSize: Int = 16): VariantDataset = {
-    requireSplit("linear regression 3")
+    requireSplit("linear regression")
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/VariantDataset.scala]: nice find

File: src/main/scala/is/hail/variant/VariantDataset.scala [hunk 4]
```diff
@@ -469,8 +469,8 @@ class VariantDatasetFunctions(private val vds: VariantDataset) extends AnyVal {
   }
 
   def logregBurden(keyName: String, variantKeys: String, singleKey: Boolean, aggExpr: String, test: String, y: String, covariates: Array[String] = Array.empty[String]): (KeyTable, KeyTable) = {
-    requireSplit("linear burden regression")
-    vds.requireSampleTString("linear burden regression")
+    requireSplit("logistic burden regression")
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/VariantDataset.scala]: good catch

File: python/hail/dataset.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/dataset.py]: or `Requires that the dataset contains`

File: python/hail/dataset.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/dataset.py]: or `requires that the ... is`

**Comment by jigold** [comment]
@tpoterba back to you.

**Pull request merged**